kube-controllers/pkg/controllers/node/ipam.go

```diff
@@ -869,6 +890,10 @@ func (c *ipamController) allocationIsValid(a *allocation, preferCache bool) bool
 }
 
 func (c *ipamController) syncIPAM() error {
+	if !c.datastoreReady {
+		return fmt.Errorf("datastore is locked, skipping ipam sync")
```

Comment:
This is going to trigger an immediate retry - I don't think we want to do that. We should just log a warning and return `nil` so that we don't tight loop